src/main/java/com/savvato/tribeapp/controllers/ConnectAPIController.java

```diff
@@ -92,10 +93,13 @@ public void connect(@Payload ConnectIncomingMessageDTO incoming, @Header("simpSe
   @PostMapping("/cosign")
   public ResponseEntity<CosignDTO> saveCosign(@RequestBody @Valid CosignRequest cosignRequest) {
 
-      CosignDTO cosignDTO = cosignService.saveCosign(cosignRequest.userIdIssuing, cosignRequest.userIdReceiving, cosignRequest.phraseId);
-      
-      return ResponseEntity.status(HttpStatus.OK).body(cosignDTO);
+    Optional<CosignDTO> opt = cosignService.saveCosign(cosignRequest.userIdIssuing, cosignRequest.userIdReceiving, cosignRequest.phraseId);
 
+    if(opt.isEmpty()) {
+      log.error("Users may not cosign themselves. ");
+      return ResponseEntity.status(HttpStatus.FORBIDDEN).body(null);
```

Comment:
According to a quick Googling, 400 BAD REQUEST, perhaps including the string "Users may not cosign themselves.", is most appropriate.

Reply:
In oder to return a message, I changed the return type from ResponseEntity<CosignDTO> to ResponseEntity. The endpoint now returns either a CosignDTO or a GenericMessageDTO with a string. Is this what you were envisioning?

Reply:
Also, I had a couple merge conflicts to fix, and it may make for some strange formatting commits in the cosign service imports. I didn't realize you were working on it and I didn't pull before push. oops.